src/Tribe/Main.php

```diff
@@ -4542,15 +4490,8 @@ public static function array_insert_after_key( $key, $source_array, $insert_arra
 		 * @return array
 		 */
 		public static function array_insert_before_key( $key, $source_array, $insert_array ) {
-			if ( array_key_exists( $key, $source_array ) ) {
-				$position     = array_search( $key, array_keys( $source_array ) );
-				$source_array = array_slice( $source_array, 0, $position, true ) + $insert_array + array_slice( $source_array, $position, null, true );
-			} else {
-				// If no key is found, then add it to the end of the array.
-				$source_array += $insert_array;
-			}
-
-			return $source_array;
+			_deprecated_function( __METHOD__, '4.0', 'Tribe__Main::array_insert_before_key' );
+			return $this->common()->array_insert_before_key( $key, $source_array, $insert_array );
```

Comment:
`$this` doesn't make sense here - we're inside an instance method at this point.